### PR TITLE
Single video play page done

### PR DESCRIPTION
**Pull request by animeshgarai09**
Hey there!
Added a few video data in backend DB, and displayed them all in explore page.
Created the single video playback page with watch-later functionality added.

Thanks for the review

### CODE REVIEW

File: src/pages/watch-later/watch-later.js [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
-
+import { useWatchLater } from "@providers/watch-later-provider"
 const WatchLater = () => {
+    const { WatchLaterState } = useWatchLater()
     return (
         <div className={styles.container}>
             <SidePane title="Watch Later..." time="30 Minutes" count={30} />
             <div className={styles.main}>
+                {
+                    WatchLaterState.length !== 0
```
Comment (foolhardy21) [inline src/pages/watch-later/watch-later.js:11]: suggestion - optional chaining can be used here for `watchLaterState.map()`. Please correct me if I'm wrong.

File: src/pages/watch-later/watch-later.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+                                videoItem={item}
+                                isRemoveWatchLater={true}
+                                key={i}
+                            />
+                        })
+                        : [...Array(4)].map((_, i) => {
+                            return <VideoCard orientation="horizontal" key={i} isLoading={true} />
+                        })
+                }
                 {/* <VideoCard orientation="horizontal" />
```
Comment (foolhardy21) [inline src/pages/watch-later/watch-later.js:24]: commented code.

File: src/pages/watch/watch.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import styles from "./watch.module.scss"
+import { VideoEmbed } from "@components"
+import { Avatar } from "react-felix-ui"
+import { useVideos } from "@providers/video-provider"
+import { useWatchLater } from "@providers/watch-later-provider"
```
Comment (foolhardy21) [inline src/pages/watch/watch.js:5]: question - can you tell me a bit about `@providers`. Have you published these like your cl ?

File: src/store/providers/video-provider.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ const VideoProvider = ({ children }) => {
         "/api/videos",
         "/api/categories",
     ];
+
     useEffect(() => {
         Promise.all(endpoints.map((endpoint) => axios.get(endpoint))).then(
```
Comment (foolhardy21) [inline src/store/providers/video-provider.js:19]: appreciation - good application of `Promise.all`

**Review by foolhardy21 — APPROVED** [review]
clean code. nice work.

File: src/pages/watch/watch.js (hunk 3 above)
Comment (animeshgarai09) [reply src/pages/watch/watch.js:5]: I have set allies for my folder path with `react-app-rewire-alias` package

File: src/pages/watch-later/watch-later.js (hunk 1 above)
Comment (animeshgarai09) [reply src/pages/watch-later/watch-later.js:11]: Not able to understand what you are suggesting here

Comment (foolhardy21) [reply src/pages/watch-later/watch-later.js:11]: I was suggesting that `watchlaterState?.map()`would also work if the state is empty. So you don't have to check for length seperately.

Comment (animeshgarai09) [reply src/pages/watch-later/watch-later.js:11]: Yeah that would work but i need else condition for empty list UI element

File: src/backend/db/videos.js [hunk 5]
```diff
@@ -88,13 +86,13 @@ export const videos = [
...
         thumbnail: "KfB2sx9uCkI-MQ.jpg",
-        url: "https://www.youtube.com/watch?v=KfB2sx9uCkI",
+        embedId: "KfB2sx9uCkI",
         views: "50k",
         uploadedOn: "1 year ago",
         category: "Documentary",
     },
     {
-        _id: uuid(),
+        _id: 'xpmy12-k8fqxp-lwaph1qj',
```
Comment (MayankKumar10) [inline src/backend/db/videos.js:95]: is uuid is not working here

File: src/components/side-nav/side-nav.js [hunk 6]
```diff
@@ -24,7 +24,7 @@ const SideNav = () => {
                     <NavLink className={(navigationData) => navigationData.isActive ? styles.active : null} to="/playlists"><IconButton icon={<RiPlayList2Fill />} className={styles.icon} /></NavLink>
                 </ListItem>
                 <ListItem className={styles.nav_link}>
-                    <NavLink className={(navigationData) => navigationData.isActive ? styles.active : null} to="/watch-later"><IconButton icon={<FiClock />} className={styles.icon} /></NavLink>
+                    <NavLink className={(navigationData) => navigationData.isActive ? styles.active : null} to="/watch-later"><IconButton icon={<AiOutlineClockCircle />} className={styles.icon} /></NavLink>
```
Comment (MayankKumar10) [inline src/components/side-nav/side-nav.js:27]: you can write it down in the arrow function that's why you don't have to repeat your code

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions else everything looks good to me you can merge it.

File: src/backend/db/videos.js (hunk 5 above)
Comment (animeshgarai09) [reply src/backend/db/videos.js:95]: uuid is not required for my functionality

File: src/components/side-nav/side-nav.js (hunk 6 above)
Comment (animeshgarai09) [reply src/components/side-nav/side-nav.js:27]: Sure will do that

**Pull request merged**